vision.py

```diff
@@ -120,68 +124,122 @@ def is_game_piece_present(
 
     Args:
-        frame (np.ndarray): New camera frame containing goals
-        regions_of_interest (List[GoalRegionObservation]): List of goal region observations with no information about occupancy
+        frame (np.ndarray): New camera frame containing nodes
+        regions_of_interest (List[NodeRegionObservation]): List of node region observations with no information about occupancy
 
     Returns:
-        List[GoalRegionObservation]: List of goal region observations
+        List[NodeRegionObservation]: List of node region observations
     """
     pass
 
 
 def annotate_image(
     frame: np.ndarray,
-    map: list[GoalState],
-    goal_observations: list[GoalRegionObservation],
+    map: list[NodeRegionState],
+    node_observations: list[NodeRegionObservation],
 ) -> np.ndarray:
-    """annotate a frame with projected goal points
+    """annotate a frame with projected node points
 
     Args:
         frame (np.ndarray): raw image frame without annotation
-        map (List[GoalState]): current map state with information on occupancy state_
-        goal_observations (List[GoalRegionObservation]): goal observations in the current time step
+        map (List[NodeState]): current map state with information on occupancy state_
+        node_observations (List[NodeRegionObservation]): node observations in the current time step
 
     Returns:
-        np.ndarray: frame annotated with observed goal regions
+        np.ndarray: frame annotated with observed node regions
     """
     pass
 
 
+def is_node_region_in_image(
+    robot_pose: Pose2d,
+    camera_params: CameraParams,
+    node_region: NodeRegion,
+) -> bool:
+
+    # create transform to make camera origin
+    world_to_robot = Transform3d(Pose3d(), Pose3d(robot_pose))
+    world_to_camera = world_to_robot + camera_params.transform
+    node_region_camera_frame = world_to_camera.inverse() + Transform3d(
+        node_region.position, Rotation3d()
+    )
+
+    # Check the robot is facing the right direction for the point by checking it is inside the FOV
+    return point3d_in_field_of_view(
+        node_region_camera_frame.translation(), camera_params
+    )
+
+
+def point3d_in_field_of_view(point: Translation3d, camera_params: CameraParams) -> bool:
+    """Determines if a point in 3d space relative to the camera coordinate frame is visible in a camera's field of view
+
+    Args:
+        point (Translation3d): _point in 3d space relative to a camera
+        camera_params (CameraParams): camera parameters structure providing information about a frame being processed
+
+    Returns:
+        bool: if point is visible
+    """
+    vertical_angle = atan2(point.z, point.x)
+    horizontal_angle = atan2(point.y, point.x)
+
+    return (
+        (point.x > 0)
+        and (
+            vertical_angle > -camera_params.get_vertical_fov() / 2
+            and vertical_angle < camera_params.get_vertical_fov() / 2
```

Comment:
```suggestion
            -camera_params.get_vertical_fov() / 2
            < vertical_angle < camera_params.get_vertical_fov() / 2
```

It'd probably be worth making a variable for the half FOV.